cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -775,6 +796,16 @@ class ReturnIndirectionInstruction extends VariableInstruction {
    * Holds if this instruction is the return indirection for `this`.
    */
   final predicate isThisIndirection() { var instanceof IRThisVariable }
+
+  /**
+   * Holds if this instruction is the return indirection for the parameter with index `index`, or
+   * if this instruction is the return indirection for `this` and `index` is `-1`.
+   */
+  final predicate hasIndex(int index) {
+    index >= 0 and index = this.getParameter().getIndex()
+    or
+    index = -1 and this.isThisIndirection()
+  }
```

Comment:
The implementation of this predicate looks a lot like `getArgumentOperand`, but only the latter has `pragma[noinline]` on it. Is there a good reason to put `pragma[noinline]` on one but not the other?

Reply:
Hm, no. I suspect the `pragma[noinline]` is an artifact of me copy/pasting from `getPositionalArgumentOperand`. I'll just double-check that I can safely remove the new `pragma[noinline]`s.

Reply:
Hm... looking at [this comment you made](https://github.com/github/codeql/pull/4637#discussion_r519897265), I'd actually feel more comfortable adding a `pragma[noinline]` to the `hasIndex ` predicates. It doesn't seem to be necessary right now, but if it avoids a future bad join-order we might as well put it there now. Would that be reasonable?

Reply:
Yes.